In [2]:
library(tidyverse)
library(repr)
library(tidymodels)
library(GGally)
library(ISLR)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.6     ✔ rsample      1.2.1
✔ dials        1.3.0     ✔ tune         1.1.2
✔ infer        1.0.7     ✔ workflows    1.1.4
✔ modeldata    1.4.0     ✔ workflowsets 1.0.1
✔ parsnip      1.2.1     ✔ yardstick    1.3.1
✔ recipes      1.1.0     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filt

## 1. Data Discription

In [16]:
players <- read_csv("data/players.csv")
sessions <- read_csv("data/sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [28]:
subscribe <- players |>
  summarise(true = sum(subscribe, na.rm = TRUE),
    false = sum(!subscribe, na.rm = TRUE),
    t = 144/(144+52),
    f = 52/(144+52))

played_hours <- players |>
  summarise(mean = round(mean(played_hours, na.rm = TRUE), 2),
    min = min(played_hours, na.rm = TRUE),
    max = max(played_hours, na.rm = TRUE))

Age <- players |>
  summarise(mean = round(mean(Age, na.rm = TRUE), 2),
    min = min(Age, na.rm = TRUE),
    max = max(Age, na.rm = TRUE),)

subscribe
played_hours
Age

true false t         f        
1 144  52    0.7346939 0.2653061

mean min max  
1 5.85 0   223.1

mean  min max
1 21.14 9   58

### 1.1 Summary for `players.csv`

General information
- Number of observations: 196

- Number of variables: 7

- Purpose: Contains player profile and activity data.


| Variable Name | Type | Description | Summary |
|----------------|------|--------------|--------------------------|
| experience | chr | Player experience level  | / |
| hashedEmail | chr | Unique identifier (hashed email) for linking datasets | / |
| name | chr | Player name | / |
| gender | chr | Player gender identity | /|
| subscribe | lgl  | Whether the player has an active subscription | 144 True (73.5%), 52 False (26.5%) |
| played_hours | dbl | Total number of hours played | Mean = 5.85, Min = 0, Max = 223.1 |
| Age | dbl | Player age | Mean = 21.14, Min = 9, Max = 58, 2 NAs |

### 1.2 Summary for `sessions.csv`

General information
- Number of observations: 1,535

- Number of variables: 5

- Purpose: Records game session information per player.


| Variable Name | Type | Description 
|----------------|------|-------------
| hashedEmail | chr | Player identifier | 
| start_time | chr | Session start time | 
| end_time | chr | Session end time | 
| original_start_time | dbl | System-recorded session start|
| original_end_time | dbl | System-recorded session end|

### 1.3 Potential issues

- Missing Data: 2 missing entries for `Age`, 2 missing `end_time` and `original_end_time` values.
- Categorical Imbalance: Most players are Male, indicate that gender may not be a good predictor